scipy/special/_cdflib.pyx

```diff
@@ -0,0 +1,7092 @@
+                S.status = 1
+                return
+
+        elif S.next_state == 200:
+
+            S.fb = S.fx
+            if not (S.fc*S.fb >= 0.):
+            # 210
+                if not (S.w == S.mb):
+            # 220
+                    S.ext += 1
+                else:
+                    S.ext = 0
+            else:
+                # 70
+                S.c = S.a
+                S.fc = S.fa
+                S.ext = 0
+            # 230
+            S.next_state = 80
+
+        elif S.next_state == 240:
+            S.xhi = S.c
+            S.qrzero = (((S.fc >= 0.) and (S.fb <= 0.)) or
+                        ((S.fc < 0.) and (S.fb >= 0.)))
+
+            if S.qrzero:
+                S.status = 0
+            else:
+            # 250
+                S.status = -1
+
+            return
+
+        else:
+            S.status = -9999  # Bad state, should not be possible to get here
+            return
+
+
+# %%-------------------------------------- erf
+cdef inline double erf(double x) noexcept nogil:
+    """
+    Evaluation of the real error function
+    """
+    cdef double ax, bot, t, top
+    cdef double c = .564189583547756
+    cdef double[5] a = [.771058495001320e-04, -.133733772997339e-02,
+                        .323076579225834e-01, .479137145607681e-01,
+                        .128379167095513e+00]
+    cdef double[3] b = [.301048631703895e-02, .538971687740286e-01,
+                        .375795757275549e+00]
+    cdef double[8] p = [-1.36864857382717e-07, 5.64195517478974e-01,
+                        7.21175825088309e+00, 4.31622272220567e+01,
+                        1.52989285046940e+02, 3.39320816734344e+02,
+                        4.51918953711873e+02, 3.00459261020162e+02]
+    cdef double[8] q = [1.00000000000000e+00, 1.27827273196294e+01,
+                        7.70001529352295e+01, 2.77585444743988e+02,
+                        6.38980264465631e+02, 9.31354094850610e+02,
+                        7.90950925327898e+02, 3.00459260956983e+02]
+    cdef double[5] r = [2.10144126479064e+00, 2.62370141675169e+01,
+                        2.13688200555087e+01, 4.65807828718470e+00,
+                        2.82094791773523e-01]
+    cdef double[4] s = [9.41537750555460e+01, 1.87114811799590e+02,
+                        9.90191814623914e+01, 1.80124575948747e+01]
+
+    ax = abs(x)
+    if ax <= 0.5:
+        t = x*x
+        top = ((((a[0]*t+a[1])*t+a[2])*t+a[3])*t+a[4]) + 1.
+        bot = ((b[0]*t+b[1])*t+b[2])*t + 1.
+        return x*(top/bot)
+
+    if ax <= 4.:
+        top = (((((((p[0]
+                    )*ax+p[1]
+                   )*ax+p[2]
+                  )*ax+p[3]
+                 )*ax+p[4]
+                )*ax+p[5]
+               )*ax+p[6]
+              )*ax + p[7]
+        bot = (((((((q[0]
+                    )*ax+q[1]
+                   )*ax+q[2]
+                  )*ax+q[3]
+                 )*ax+q[4]
+                )*ax+q[5]
+               )*ax+q[6])*ax + q[7]
```

Comment:
There are a lot of lines like this doing Horner's method, if I remember correctly @steppi added a function for this using one of the Knuth algorithms in one of his recent PRs. Could this be used here and elsewhere in this code? Equally happy for that to be left to a follow up

Reply:
That's close. I recently translated an existing Cython implementation that works for complex values into C++. For real values like this we can use `polevl` from cephes (that's what I've been doing so far in the Cython->C++ rewrites), but I think it's fine to leave stuff like this for a follow-up.

Reply:
Since these are super old functions I think noone will read them (and probably directly replace them, in case of improvements). Not sure it is worth the effort for the surgery. I'm going to rewrite them in C anyways. So this is just a stepping stone to remove the Fortran parts. Then we can do all kinds of improvements.